www/scripts/codecheckerviewer/filterHelper.js

```diff
@@ -80,93 +89,135 @@ function (declare, Button, ContentPane, Filter) {
     }
-    else
-      CC_SERVICE.getDiffResultTypes(
-        run1,
-        run2,
-        difftype,
-        [reportFilter],
-        result);
   }
 
-  function addPathFilterField(filter) {
-    filter.addField('text', 'path', 'Path filter');
+  /**
+   * Add path field and set to the value.
+   */
+  function addPathFilterField(filter, value) {
+      filter.addField('text', 'path', 'Path filter', value);
   }
 
-  function addSuppressionStateField(filter) {
-    filter.addField('select', 'suppression', [
-      { label : 'Unsuppressed', value : 'unsup' },
-      { label : 'Suppressed',   value : 'supp'  }
-    ]);
+  /**
+   * Add suppression field and select one value.
+   */
+  function addSuppressionStateField(filter, selected_value) {
+
+    if (selected_value === undefined){
+      selected_value = 'unsup'
+    }
+
+    var options = [
+      { label : 'Unsuppressed', value : 'unsup', selected: 'unsup' == selected_value},
+      { label : 'Suppressed',   value : 'supp', selected: 'supp' == selected_value}
+    ]
+
+    filter.addField('select', 'suppression', options);
   }
 
-  function addCheckerIndexField(filter, run1, run2) {
+  /**
+   * Register the callbacks for the checker filter fields.
+   */
+  function addCheckerIndexField(filter, filterValues, run1, run2) {
     var callback
-      = arguments.length === 2
-      ? function (cb) { return checkerInfoOptions(filter, run1, cb); }
-      : function (cb) { return checkerInfoOptions(filter, run1, run2, cb); }
-
+      = arguments.length === 3
+      ? function (cb) { return checkerInfoOptions(filter, filterValues, run1, cb); }
+      : function (cb) { return checkerInfoOptions(filter, filterValues, run1, run2, cb); }
     filter.addField('select', 'checker', callback);
   }
 
-  function addDiffTypeField(filter) {
-    filter.addField('select', 'difftype', [
+  /**
+   * Set the diff type filter field.
+   */
+  function addDiffTypeField(filter, selected_value) {
+    var options = [
       { label : 'New only',   value : 'new'        },
-      { label : 'Resolved',   value : 'resolved'   },
-      { label : 'Unresolved', value : 'unresolved' }
-    ]);
+      { label : 'Resolved',   value : 'resolved', selected: 'resolved' == selected_value },
+      { label : 'Unresolved', value : 'unresolved', selected: 'unresolved' == selected_value}
+    ];
+
+    filter.addField('select', 'difftype', options);
   }
 
+  /**
+   * Create a filter group which can contain multiple filters.
+   */
   var FilterGroup = declare(ContentPane, {
-    constructor : function () {
+    constructor : function (filters) {
       this.style = 'padding: 5px';
     },
 
     postCreate : function () {
-      this._addPlusButton(this._addFilter());
+        filters = this.filters;
+        if (filters === undefined) {
+          this._addPlusButton(this._addFilter());
+        } else {
+          this._addPlusButton(this._addFilter(filters[0]));
+          for (i = 1; i < filters.length; i++) {
```

Comment:
Use array forEach instead of for loop:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach